src/modify.js

```diff
@@ -60,3 +60,37 @@ exports.modifyService = function(env, serviceName, url, remove) {
     })
   })
 }
+
+exports.modifyMultipleServices = function(env, newImports) {
+  return new Promise((resolve, reject) => {
+    lock.writeLock(releaseLock => {
+      const resultPromise = ioOperations.readManifest(env)
+        .then(data => {
+          let json
+          try {
+            json = data ? JSON.parse(data) : getEmptyManifest()
+          } catch (err) {
+            releaseLock()
+            reject('Could not read import map -- ' + err)
```

Comment:
I'm probably wrong here, but if you just `reject()` here, it should go to the `.catch` block which will release the lock right? so no need to call `releaseLock` here?

Reply:
actually the `reject()` function is for the outermost promise that is returned from the modifyMultipleServices function. It's not for the `resultPromise` but for the `writeLockPromise`. So it would not go into the catch block for the resultPromise.

However, looking this over makes me realize that it's quite confusing -- we should probably just let the error be thrown instead of catching it, so that the resultPromise's `.catch()` block picks it up and does the releaseLock() call. I'll change to that.